src/nu/validator/htmlparser/impl/TreeBuilder.java

```diff
@@ -438,6 +438,8 @@
 
     private boolean reportingDoctype = true;
 
+    private HashMap<String, String> errorProfileMap = null;
```

Comment:
Is this necessary? Can this be obtained from the tokenizer (even though it only exists on a subclass of the tokenizer)?

Reply:
```java
    private HashMap<String, String> errorProfileMap = null;
```

> Is this necessary? Can this be obtained from the tokenizer (even though it only exists on a subclass of the tokenizer)?

I added a `getErrorProfile()` method to the tokenizer. But I don’t see how to make it to compiled if it’s only a method of the ErrorReportingTokenizer subclass — so I also made it a method of the Tokenizer superclass, and then overrode it in the ErrorReportingTokenizer subclass.

Is there some way I could instead make it work if it only exists in the ErrorReportingTokenizer subclass (so that I don’t have to also clutter the superclass with it)?

Also, did you have in mind that a getter isn’t really needed here, but instead I could just make the errorProfileMap of the tokenizer protected (instead of private), and then just have the treebuilder code use that (rather then needing a getter)?